docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/StochasticGradientDescentNonCalibrated.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public static class StochasticGradientDescentNonCalibrated
+    {
+        // In this examples we will use the adult income dataset. The goal is to predict
+        // if a person's income is above $50K or not, based on demographic information about that person.
+        // For more details about this dataset, please see https://archive.ics.uci.edu/ml/datasets/adult.
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this example to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
+
+            // Download and featurize the dataset.
+            var data = SamplesUtils.DatasetUtils.LoadFeaturizedAdultDataset(mlContext);
```

Comment:
It'd nice if you can use in-memory data for trainers. Such a way has some benefits as discussed in #2726.

Reply:
Hi Wei-Sheng, please mark your in-memory comments as optional, until #2726 reaches a conclusion. #2726 is still in discussion and there's no clear actionable change that everybody agrees on, yet. 

---
In reply to: [261020122](https://github.com/dotnet/machinelearning/pull/2770#discussion_r261020122) [](ancestors = 261020122)

Reply:
Lets do these things once we have decision. 

---
In reply to: [261770052](https://github.com/dotnet/machinelearning/pull/2770#discussion_r261770052) [](ancestors = 261770052,261020122)